cypress/integration/pages/onDemandTV/tests.js

```diff
@@ -39,5 +42,81 @@ export default ({ service, pageType, variant, isAmp }) => {
         });
       });
     });
+    describe(`Tests for ${service} ${pageType} ${variant} with toggle use`, () => {
+      before(() => {
+        cy.getToggles(config[service].name);
+      });
+      describe('Recent Episodes component', () => {
+        it('should be displayed if the toggle is on, and shows the expected number of items (does not run on local)', function test() {
+          if (Cypress.env('APP_ENV') === 'local') {
+            cy.log('Does not run on local');
+          } else {
+            cy.fixture(`toggles/${config[service].name}.json`).then(toggles => {
+              const recentEpisodesEnabled = path(
+                ['recentVideoEpisodes', 'enabled'],
+                toggles,
+              );
+              cy.log(
+                `Recent Episodes component enabled? ${recentEpisodesEnabled}`,
+              );
+
+              if (recentEpisodesEnabled) {
+                // There cannot be more episodes shown than the max allowed
+                const recentEpisodesMaxNumber = path(
+                  ['recentVideoEpisodes', 'value'],
+                  toggles,
+                );
+                cy.log(`Recent Episodes max limit? ${recentEpisodesMaxNumber}`);
+                const currentPath = Cypress.env('currentPath');
+                const url =
+                  Cypress.env('APP_ENV') === 'test'
+                    ? `${currentPath}?renderer_env=live`
+                    : `${currentPath}`;
+
+                cy.request(getDataUrl(url)).then(({ body }) => {
+                  // Count the number of episodes that are available and so will show (there can be unavailable episodes in the list)
+                  const expectedNumberOfEpisodes = body.relatedContent.groups[0].promos
+                    .filter(({ media }) => media.versions.length)
+                    .slice(0, recentEpisodesMaxNumber).length;
+
+                  cy.log(
+                    `Number of available episodes? ${expectedNumberOfEpisodes}`,
+                  );
+                  if (expectedNumberOfEpisodes > 1) {
+                    cy.get("ul[class*='css-1ddpce6-StyledEpisodeList']").should(
+                      'exist',
+                    );
+
+                    cy.get("ul[class*='css-1ddpce6-StyledEpisodeList']").within(
```

Comment:
As above with the selectors being too specific, and to investigate whether a lot of this is shared logic between TV/radio and could be separated into its own test file?